26 implement player and playerlist

stephensmeal opened the pull request:
Requesting peer review of the following classes to be merged into the integ branch:
Player, PlayerList, and PlayerStatus

patmay commented:
@yo09975 why are all of the files from integ showing up as new files?

stephensmeal committed:
Recommitting Player Assoc Classes To Fix Branch Issue

Recommitting the following files since the past committ was done
on an incorrect branch.
player.py
playerlist.py
playerstatus.py

patmay commented:
get_next_player and get_next_turn should iterate through the playerlist, checking player status (== ACTIVE for _turn, == ACTIVE or LOST for player) until it finds one of the correct status. get_next_turn should return None if there are no other ACTIVE players. If a bad reference index is provided (current returns None), then an error should be thrown.

stephensmeal committed:
Fixed PlayerList Next Turn and Next player Methods

Additional work to PlayerList's get_next_turn and get_next_player
methods in order to account for Player statuses.
playerlist.py
